08_education/programs/MEPS_metric_city.qmd

```diff
@@ -0,0 +1,378 @@
+
+### School level enrollment
+
+Download the CCD school level enrollment by race/ethnicity from Urban's Education Data Portal
+Filter grade for 99 - 99 is the total enrollment in the school, regardless of grade.
+
+Note this download is computationally large, will take time to finish.
+
+```{r}
+if(!file.exists(here::here("08_education/data/raw/", paste0("ccd_enr_2014-", year_ed, ".rds"))))
+{
+  education_data <- get_education_data(level = 'schools', 
+                         source = 'ccd', 
+                         topic = 'enrollment', 
+                         subtopic = 'race',
+                         filters = list(year = 2014:year_ed,
+                                        grade = 99),
+                         add_labels = TRUE)
+  write_rds(education_data, 
+            here::here("08_education/data/raw/", paste0("ccd_enr_2014-", year_ed, ".rds")))
+}  else {
+    education_data <- read_rds(here::here("08_education/data/raw/", paste0("ccd_enr_2014-", year_ed, ".rds")))
+}
+
+```
+
+Add the number of enrolled male and female students within race
+
+```{r}
+education_data <- education_data %>% 
+  select(-grade) %>% 
+  filter(sex != 99) %>% 
+  mutate(enrollment = ifelse(enrollment < 0, NA_integer_, enrollment)) %>% 
+  group_by(year, ncessch, ncessch_num, leaid, fips, race) %>% 
+  summarise(enrollment = sum(enrollment, na.rm = TRUE))
+```
+
+Reshape the the data wide by race
+
+```{r}
+
+ed_data_wide <- education_data %>% 
+  pivot_wider(id_cols = c(year, ncessch), names_from = race, values_from = enrollment) 
+
+```
+
+### CCD district data
+
+Download district data from CCD, includes total enrollment, city location and county codes
+
+```{r}
+
+if(!file.exists(here::here("08_education/data/raw/", paste0("ccd_dir_2014-", year_ed, ".rds"))))
+{
+  ccd_directory <- get_education_data(level = 'schools', 
+                         source = 'ccd', 
+                         topic = 'directory',
+                         filters = list(year = 2014:year_ed),
+                         add_labels = FALSE)
+  write_rds(ccd_directory, 
+            here::here("08_education/data/raw/", paste0("ccd_dir_2014-", year_ed, ".rds")))
+}  else {
+    ccd_directory <- read_rds(here::here("08_education/data/raw/", paste0("ccd_dir_2014-", year_ed, ".rds")))
+}
+
+```
+
+### Meps data
+
+MEPS data
+
+```{r}
+year_meps <- 2020
+if(!file.exists(here::here("08_education/data/raw/", paste0("ccd_meps_2014-", year_meps, ".rds"))))
+{
+  meps <- get_education_data(level = 'schools', 
+                         source = 'meps', 
+                         filters = list(year = 2014:year_meps))
+  write_rds(meps, 
+            here::here("08_education/data/raw/", paste0("ccd_meps_2014-", year_meps, ".rds")))
+}  else {
+    meps <- read_rds(here::here("08_education/data/raw/", paste0("ccd_meps_2014-", year_meps, ".rds")))
+}
+
+```
+
+## Data merge
+
+```{r}
+
+ccd_directory <- ccd_directory %>% 
+  mutate(enrollment = if_else(enrollment < 0, 0, enrollment))
```

Comment:
Why is enrollment negative?

Reply:
The negative values are codes: https://educationdata.urban.org/documentation/schools.html#ccd-enrollment-by-grade